engine/src/test/java/io/zeebe/engine/processor/workflow/multiinstance/MultiInstanceActivityTest.java

```diff
@@ -518,6 +521,48 @@ public void shouldIterateOverNestedInputCollection() {
         .containsExactlyElementsOf(INPUT_COLLECTION);
   }
 
+  @Test
+  public void shouldCollectNestedOutputElements() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            workflow(
+                miBuilder.andThen(
+                    m -> m.zeebeOutputElementExpression(OUTPUT_ELEMENT_EXPRESSION + ".nested"))))
+        .deploy();
+
+    final long workflowInstanceKey =
+        ENGINE
+            .workflowInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariable(INPUT_COLLECTION_EXPRESSION, INPUT_COLLECTION)
+            .create();
+
+    RecordingExporter.jobRecords(JobIntent.CREATED)
+        .withWorkflowInstanceKey(workflowInstanceKey)
+        .limit(INPUT_COLLECTION.size())
+        .exists();
+
+    // complete job
+    completeJobs(
+        workflowInstanceKey,
+        INPUT_COLLECTION.size(),
+        i -> Map.of("nested", OUTPUT_COLLECTION.get(i)));
+
+    // then
+    final List<Long> scopeKeys =
+        RecordingExporter.variableRecords(VariableIntent.CREATED)
+            .withName(OUTPUT_ELEMENT_EXPRESSION)
+            .map(Record::getValue)
+            .map(VariableRecordValue::getScopeKey)
+            .limit(3)
+            .collect(Collectors.toList());
+    assertThat(scopeKeys)
+        .hasSize(3)
+        .allSatisfy(scopeKey -> assertThat(scopeKeys).containsOnlyOnce(scopeKey));
```

Comment:
What do you want to check here?

I think that we should verify that 
* the variable is created without `.nested` in its name (which is the issue that was fixed)
* the variable is initialized with `null`
* the values are collected in the output collection